### ISSUE TITLE
macOS: issues with wb-engines

### ISSUE THREAD

**Issue by twoplan** [nguyenpham/Banksia#6]
A small tournament with 7 wb-engines (no opening book) on Banksia 2.5 
```
 "players" : 
  [
    "EXchess v7.97b",
    "Crafty-25.3",
    "Phalanx XXV",
    "OliThink 5.3.3",
    "TSCP 1.81",
    "FrankWalter 2.4.0",
    "Scorpio_2.9.0 MCTS+NN"
  ],
```
Only Crafty-25.3, FrankWalter 2.4.0 and Phalanx XXV played regular games. 

- The other 4 had all `[Termination 	"timeout"]` in the pgn. **These engines never moved!**

- Interrupting the tournament with CTRL-C left **engines running in the background** (e.g. OliThink and EXchess). Each hanging engine used 100% cpu from one core.

EXchess log:

```
21:43:30 banksia> 
1. Crafty-25.3 vs EXchess v7.97b

21:43:30 Crafty-25.3< xboard
protover 2
21:43:30 EXchess v7.97b< xboard
protover 2
21:43:31 Crafty-25.3> hash table memory = 512M bytes (32M entries).
21:43:31 EXchess v7.97b> feature setboard=1
21:43:31 EXchess v7.97b< accepted setboard
21:43:31 EXchess v7.97b> feature playother=1
21:43:31 EXchess v7.97b< accepted playother
21:43:31 EXchess v7.97b> feature usermove=1
21:43:31 EXchess v7.97b< accepted usermove
21:43:31 EXchess v7.97b> feature ics=1
21:43:31 EXchess v7.97b< accepted ics
21:43:31 EXchess v7.97b> feature smp=1
21:43:31 EXchess v7.97b< accepted smp
21:43:31 EXchess v7.97b> feature memory=1
21:43:31 EXchess v7.97b< accepted memory
21:43:31 EXchess v7.97b> feature variants="normal,nocastle,fischerandom"
21:43:31 EXchess v7.97b< accepted variants
21:43:31 EXchess v7.97b> feature myname="EXchess v7.97b"
21:43:31 EXchess v7.97b< accepted myname
21:43:31 EXchess v7.97b> feature option="Playing Strength -slider 100 1 100
21:43:31 EXchess v7.97b> feature done=1
21:43:31 Crafty-25.3> pawn hash table memory = 48M bytes (2M entries).
21:43:31 Crafty-25.3> parallel threads disabled.
21:43:31 Crafty-25.3> SMP terminate extra threads when idle.
21:43:31 Crafty-25.3> SYZYGY EGTB access disabled.
21:43:31 Crafty-25.3> pondering disabled.
21:43:31 Crafty-25.3> book learning disabled
21:43:31 Crafty-25.3> Crafty v25.3
21:43:31 Crafty-25.3> machine has 4 processors
21:43:31 Crafty-25.3> White(1):
21:43:31 Crafty-25.3> tellicsnoalias set 1 Crafty v25.3 (1 cpus)
21:43:31 Crafty-25.3> tellicsnoalias kibitz Hello from Crafty v25.3! (1 cpus)
21:43:31 Crafty-25.3> feature ping=1 setboard=1 san=1 time=1 draw=1
21:43:31 Crafty-25.3< accepted ping
21:43:31 Crafty-25.3< accepted setboard
21:43:31 Crafty-25.3< accepted san
21:43:31 Crafty-25.3< accepted time
21:43:31 Crafty-25.3< accepted draw
21:43:31 Crafty-25.3> feature sigint=0 sigterm=0 reuse=0 analyze=1
21:43:31 Crafty-25.3< accepted sigint
21:43:31 Crafty-25.3< accepted sigterm
21:43:31 Crafty-25.3< accepted reuse
21:43:31 Crafty-25.3< accepted analyze
21:43:31 Crafty-25.3> feature myname="Crafty-25.3" name=1
21:43:31 Crafty-25.3< accepted myname
21:43:31 Crafty-25.3< accepted name
21:43:31 Crafty-25.3> feature playother=1 colors=0 memory=0
21:43:31 Crafty-25.3< accepted playother
21:43:31 Crafty-25.3< accepted colors
21:43:31 Crafty-25.3< accepted memory
21:43:31 Crafty-25.3> feature smp=0
21:43:31 Crafty-25.3< accepted smp
21:43:31 Crafty-25.3> feature variants="normal,nocastle"
21:43:31 Crafty-25.3< accepted variants
21:43:31 Crafty-25.3> feature done=1
21:43:31 EXchess v7.97b< memory 128
cores 1
21:43:31 EXchess v7.97b< easy
21:43:31 EXchess v7.97b< post
21:43:31 EXchess v7.97b< new
21:43:31 EXchess v7.97b< level 40 0:2 0
21:43:32 EXchess v7.97b< ping 1
21:43:32 EXchess v7.97b> Error (unknown command): ping
21:43:32 Crafty-25.3< easy
21:43:32 Crafty-25.3< post
21:43:32 Crafty-25.3< level 40 0:2 0
21:43:32 Crafty-25.3< ping 1
21:43:32 Crafty-25.3> pong 1
21:43:33 Crafty-25.3< time 200
otim 200
21:43:33 Crafty-25.3< go
21:43:33 Crafty-25.3> move d4
21:43:33 EXchess v7.97b< force
21:43:33 EXchess v7.97b< usermove d2d4
21:43:33 EXchess v7.97b> Got Usermove!
21:43:36 banksia> Timeleft for Crafty-25.3: 2.50, EXchess v7.97b: 2.00, used: 3.00
21:43:36 EXchess v7.97b< force
21:43:36 Crafty-25.3< force
21:43:36 banksia> 1) Crafty-25.3 vs EXchess v7.97b, #1, 1-0 (timeout)
21:43:37 EXchess v7.97b< quit
21:43:37 Crafty-25.3< quit

```
Scorpio log:

```
6. Scorpio_2.9.0 MCTS+NN vs EXchess v7.97b

21:44:04 Scorpio_2.9.0 MCTS+NN> feature done=0
21:44:04 Scorpio_2.9.0 MCTS+NN> ht 16777216 X 16 = 256.0 MB
21:44:04 Scorpio_2.9.0 MCTS+NN> eht 2097152 X 8 = 32.0 MB
21:44:04 Scorpio_2.9.0 MCTS+NN> pht 262144 X 24 = 6.0 MB
21:44:04 Scorpio_2.9.0 MCTS+NN> treeht 107373568 X 10 = 1024.0 MB
21:44:04 Scorpio_2.9.0 MCTS+NN> processors [1]
21:44:04 Scorpio_2.9.0 MCTS+NN> EgbbProbe 4.3 by Daniel Shawul
21:44:04 Scorpio_2.9.0 MCTS+NN> egbb_cache 4084 X 8216 = 32.0 MB
21:44:05 Scorpio_2.9.0 MCTS+NN> Loading egbbs....
180 egbbs loaded !
21:44:05 Scorpio_2.9.0 MCTS+NN> loading_time = 0s
21:44:05 Scorpio_2.9.0 MCTS+NN> feature name=1 myname="Scorpio_2.9.0 MCTS+NN"
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted name
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted myname
21:44:05 Scorpio_2.9.0 MCTS+NN> feature sigint=0 sigterm=0
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted sigint
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted sigterm
21:44:05 Scorpio_2.9.0 MCTS+NN> feature setboard=1 draw=0 colors=0
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted setboard
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted draw
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted colors
21:44:05 Scorpio_2.9.0 MCTS+NN> feature smp=0 memory=0
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted smp
21:44:05 Scorpio_2.9.0 MCTS+NN< accepted memory
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="log -check 0"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="clear_hash -button"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="resign -spin 900 100 30000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="cores -spin 1 1 512"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="ht -spin 256 1 131072"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="eht -spin 32 1 16384"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="pht -spin 8 1 256"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="egbb_path -path /Users/max/chess/Daten/TBS/Scorpio/"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="nn_path -path /Users/max/chess/Scorpio/net/net-maddex.uff"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="egbb_cache_size -spin 32 1 16384"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="egbb_load_type -spin 3 0 3"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="egbb_depth_limit -spin 6 0 64"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="egbb_ply_limit_percent -spin 75 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="nn_cache_size -spin 200 1 16384"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="n_devices -spin 1 1 128"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="device_type -combo *CPU /// GPU "
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="delay -spin 1 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="float_type -combo FLOAT /// *HALF  /// INT8 "
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="nn_type -spin 0 0 10"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="smp_type -combo *YBW /// ABDADA /// SHT "
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="smp_depth -spin 8 1 10"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="use_singular -check 1"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="use_probcut -check 0"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="singular_margin -spin 32 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="probcut_margin -spin 195 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="aspiration_window -spin 10 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="cpuct_init -spin 150 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="cpuct_base -spin 19652 0 100000000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="policy_temp -spin 200 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="noise_alpha -spin 30 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="noise_beta -spin 100 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="noise_frac -spin 25 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="noise_ply -spin 30 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="fpu_red -spin 33 -1000 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="fpu_is_loss -check 0"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="reuse_tree -check 1"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="backup_type -combo MINMAX /// AVERAGE /// MIX /// MINMAX_MEM /// AVERAGE_MEM /// MIX_MEM /// CLASSIC /// *MIX_VISIT"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="frac_alphabeta -spin 0 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="frac_freeze_tree -spin 100 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="frac_abrollouts -spin 20 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="frac_abprior -spin 0 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="mcts_strategy_depth -spin 30 0 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="alphabeta_depth -spin 16 1 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="evaluate_depth -spin 0 -4 100"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="virtual_loss -spin 1 0 1000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="visit_threshold -spin 800 0 1000000"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="montecarlo -check 1"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature option="treeht -spin 1023 0 131072"
21:44:05 Scorpio_2.9.0 MCTS+NN> feature done=1
21:44:05 EXchess v7.97b> feature setboard=1
21:44:05 EXchess v7.97b< accepted setboard
21:44:05 EXchess v7.97b> feature playother=1
21:44:05 EXchess v7.97b< accepted playother
21:44:05 EXchess v7.97b> feature usermove=1
21:44:05 EXchess v7.97b< accepted usermove
21:44:05 EXchess v7.97b> feature ics=1
21:44:05 EXchess v7.97b< accepted ics
21:44:05 EXchess v7.97b> feature smp=1
21:44:05 EXchess v7.97b< accepted smp
21:44:05 EXchess v7.97b> feature memory=1
21:44:05 EXchess v7.97b< accepted memory
21:44:05 EXchess v7.97b> feature variants="normal,nocastle,fischerandom"
21:44:05 EXchess v7.97b< accepted variants
21:44:05 EXchess v7.97b> feature myname="EXchess v7.97b"
21:44:05 EXchess v7.97b< accepted myname
21:44:05 EXchess v7.97b> feature option="Playing Strength -slider 100 1 100
21:44:05 EXchess v7.97b> feature done=1
21:44:05 EXchess v7.97b< memory 128
cores 1
21:44:05 EXchess v7.97b< easy
21:44:05 EXchess v7.97b< post
21:44:05 EXchess v7.97b< new
21:44:05 EXchess v7.97b< level 40 0:2 0
21:44:05 EXchess v7.97b< ping 1
21:44:05 EXchess v7.97b> Error (unknown command): ping
21:44:05 Scorpio_2.9.0 MCTS+NN< easy
21:44:05 Scorpio_2.9.0 MCTS+NN< post
21:44:05 Scorpio_2.9.0 MCTS+NN< new
21:44:05 Scorpio_2.9.0 MCTS+NN< level 40 0:2 0
21:44:05 Scorpio_2.9.0 MCTS+NN< ping 1
21:44:05 Scorpio_2.9.0 MCTS+NN> Error (unknown command): ping
21:44:05 Scorpio_2.9.0 MCTS+NN> Error (unknown command): 1
21:44:10 banksia> Timeleft for Scorpio_2.9.0 MCTS+NN: 2.00, used: 3.00, EXchess v7.97b: 2.00
21:44:10 EXchess v7.97b< force
21:44:10 Scorpio_2.9.0 MCTS+NN< force
21:44:10 banksia> 6) Scorpio_2.9.0 MCTS+NN vs EXchess v7.97b, #0, 0-1 (timeout)
21:44:11 EXchess v7.97b< quit
21:44:11 Scorpio_2.9.0 MCTS+NN< quit

```

Olithink log:

```
12. Phalanx XXV vs OliThink 5.3.3

21:46:26 OliThink 5.3.3< xboard
protover 2
21:46:26 Phalanx XXV< xboard
protover 2
21:46:26 Phalanx XXV> Phalanx XXV
21:46:26 Phalanx XXV> primary book ./pbook.phalanx, 48618 bytes
21:46:26 Phalanx XXV> secondary book ./sbook.phalanx, 263970 bytes
21:46:26 Phalanx XXV> eco file ./eco.phalanx, 171176 bytes
21:46:26 Phalanx XXV> tellics set 1 Phalanx XXV, 820 kB hashtable, 47/257 kB P/S opening book
21:46:26 Phalanx XXV> [ white, 1 ]
21:46:26 Phalanx XXV> xboard mode on
21:46:26 Phalanx XXV> feature myname="Phalanx XXV"
21:46:26 Phalanx XXV< accepted myname
21:46:26 Phalanx XXV> feature analyze=1 setboard=1 sigint=1 time=1 memory=1 draw=0 option="Randomizer (0-50) -slider 0 0 50" ping=1
21:46:26 Phalanx XXV< accepted analyze
21:46:26 Phalanx XXV< accepted setboard
21:46:26 Phalanx XXV< accepted sigint
21:46:26 Phalanx XXV< accepted time
21:46:26 Phalanx XXV< accepted memory
21:46:26 Phalanx XXV< accepted draw
21:46:26 Phalanx XXV< accepted ping
21:46:26 Phalanx XXV> feature done=1
21:46:26 OliThink 5.3.3> feature setboard=1 myname="OliThink 5.3.3" colors=0 analyze=0 done=1
21:46:26 OliThink 5.3.3< accepted setboard
21:46:26 OliThink 5.3.3< accepted myname
21:46:26 OliThink 5.3.3< accepted colors
21:46:26 OliThink 5.3.3< accepted analyze
21:46:27 OliThink 5.3.3< easy
21:46:27 OliThink 5.3.3< post
21:46:27 OliThink 5.3.3< new
21:46:27 OliThink 5.3.3< level 40 0:2 0
21:46:27 OliThink 5.3.3< ping 1
21:46:27 OliThink 5.3.3> UNKNOWN COMMAND: ping 1
21:46:27 Phalanx XXV< memory 128
21:46:27 Phalanx XXV< easy
21:46:27 Phalanx XXV< post
21:46:27 Phalanx XXV< new
21:46:27 Phalanx XXV< level 40 0:2 0
21:46:27 Phalanx XXV> pondering off
21:46:27 Phalanx XXV> post on
21:46:27 Phalanx XXV> initial position set, machine plays black
21:46:27 Phalanx XXV> level: 40 moves in 0:02, increment 0 seconds
21:46:27 Phalanx XXV< ping 1
21:46:27 Phalanx XXV> pong 1
21:46:29 Phalanx XXV< time 200
otim 200
21:46:29 Phalanx XXV< go
21:46:29 Phalanx XXV> 0      0     0        0  (e2-e4  d2-d4  c2-c4  Ng1-f3  , book1)
21:46:29 Phalanx XXV> move d2d4
21:46:29 OliThink 5.3.3< force
21:46:29 OliThink 5.3.3< d2d4
21:46:32 banksia> Timeleft for Phalanx XXV: 2.50, OliThink 5.3.3: 2.00, used: 3.00
21:46:32 OliThink 5.3.3< force
21:46:32 Phalanx XXV< force
21:46:32 Phalanx XXV> you play both
21:46:32 banksia> 12) Phalanx XXV vs OliThink 5.3.3, #1, 1-0 (timeout)
21:46:33 OliThink 5.3.3< quit
21:46:33 Phalanx XXV< quit
```

TSCP log:

```
13. TSCP 1.81 vs Phalanx XXV

21:46:33 Phalanx XXV> Phalanx XXV
21:46:33 TSCP 1.81> Tom Kerrigan's Simple Chess Program (TSCP)
21:46:33 Phalanx XXV> primary book ./pbook.phalanx, 48618 bytes
21:46:33 TSCP 1.81> version 1.81b, 3/10/16
21:46:33 Phalanx XXV> secondary book ./sbook.phalanx, 263970 bytes
21:46:33 TSCP 1.81> Copyright 2016 Tom Kerrigan
21:46:33 Phalanx XXV> eco file ./eco.phalanx, 171176 bytes
21:46:33 TSCP 1.81> "help" displays a list of commands.
21:46:33 Phalanx XXV> tellics set 1 Phalanx XXV, 820 kB hashtable, 47/257 kB P/S opening book
21:46:33 TSCP 1.81> Opening book missing.
21:46:33 Phalanx XXV> [ white, 1 ]
21:46:33 TSCP 1.81> tscp>
21:46:33 Phalanx XXV> xboard mode on
21:46:33 TSCP 1.81> Error (unknown command): protover
21:46:33 Phalanx XXV> feature myname="Phalanx XXV"
21:46:33 Phalanx XXV< accepted myname
21:46:33 Phalanx XXV> feature analyze=1 setboard=1 sigint=1 time=1 memory=1 draw=0 option="Randomizer (0-50) -slider 0 0 50" ping=1
21:46:33 Phalanx XXV< accepted analyze
21:46:33 Phalanx XXV< accepted setboard
21:46:33 Phalanx XXV< accepted sigint
21:46:33 Phalanx XXV< accepted time
21:46:33 Phalanx XXV< accepted memory
21:46:33 Phalanx XXV< accepted draw
21:46:33 Phalanx XXV< accepted ping
21:46:33 Phalanx XXV> feature done=1
21:46:34 TSCP 1.81< force
21:46:34 TSCP 1.81< ping 1
21:46:34 TSCP 1.81> Error (unknown command): ping
21:46:34 Phalanx XXV< memory 128
21:46:34 Phalanx XXV< easy
21:46:34 Phalanx XXV< post
21:46:34 Phalanx XXV< new
21:46:34 Phalanx XXV< level 40 0:2 0
21:46:34 Phalanx XXV> pondering off
21:46:34 Phalanx XXV> post on
21:46:34 Phalanx XXV> initial position set, machine plays black
21:46:34 Phalanx XXV> level: 40 moves in 0:02, increment 0 seconds
21:46:35 Phalanx XXV< ping 1
21:46:35 Phalanx XXV> pong 1
21:46:40 banksia> Timeleft for TSCP 1.81: 2.00, used: 3.00, Phalanx XXV: 2.00
21:46:40 Phalanx XXV< force
21:46:40 Phalanx XXV> you play both
21:46:40 TSCP 1.81< force
21:46:40 banksia> 13) TSCP 1.81 vs Phalanx XXV, #0, 0-1 (timeout)
21:46:41 Phalanx XXV< quit
21:46:41 TSCP 1.81< quit
```

**Comment by nguyenpham** [comment]
Thanks for the report.

I have just released v2.51 which could:
- detect more engines, including ones using scripts to start
- be workable with some engines you mentioned above (Banksia has a bug of sending pings to old-style wb engines).

Some engines are still not workable since they are not compliant to WB protocol. For example, both Phalanx XXV and OliThink 5.3.2 may (not always) start computing and produce moves before receiving the command "go" from the interface.

I don't have experience about them (non-compliant engines) but heart that they may need special treated (such as be fixed).

**Comment by twoplan** [comment]
Great, version 2.51 worked well! 
```
./banksia -jsonpath tour.json
Banksia, Chess Tournament Manager, by Nguyen Pham - version 2.51
opening books loaded, total items: 0, selection type: allnew
type: roundrobin, timer: 40/2:0, players: 7, matches: 21, concurrency: 1, ponder: off, book: off
 pgn: /Users/max/chess/Banksia/games.pgn, on
 result: /Users/max/chess/Banksia/resultlog.txt, on
 engines: /Users/max/chess/Banksia/enginelog.txt, on

1. Crafty-25.3 vs EXchess v7.97b
1) Crafty-25.3 vs EXchess v7.97b, #173, 1-0 (mate)
2. EXchess v7.97b vs Phalanx XXV
2) EXchess v7.97b vs Phalanx XXV, #119, 1-0 (mate)
3. OliThink 5.3.3 vs EXchess v7.97b
3) OliThink 5.3.3 vs EXchess v7.97b, #173, 1-0 (mate)
4. EXchess v7.97b vs TSCP 1.81
4) EXchess v7.97b vs TSCP 1.81, #85, 1-0 (mate)
5. EXchess v7.97b vs FrankWalter 2.4.0
5) EXchess v7.97b vs FrankWalter 2.4.0, #129, 1-0 (mate)
6. EXchess v7.97b vs Scorpio_2.9.0 MCTS+NN
6) EXchess v7.97b vs Scorpio_2.9.0 MCTS+NN, #51, 1-0 (mate)
7. Phalanx XXV vs Crafty-25.3
7) Phalanx XXV vs Crafty-25.3, #259, 1/2-1/2 (fifty moves)
8. OliThink 5.3.3 vs Crafty-25.3
8) OliThink 5.3.3 vs Crafty-25.3, #46, 0-1 (mate)
9. TSCP 1.81 vs Crafty-25.3
9) TSCP 1.81 vs Crafty-25.3, #48, 0-1 (mate)
10. Crafty-25.3 vs FrankWalter 2.4.0
10) Crafty-25.3 vs FrankWalter 2.4.0, #89, 1-0 (mate)
11. Crafty-25.3 vs Scorpio_2.9.0 MCTS+NN
11) Crafty-25.3 vs Scorpio_2.9.0 MCTS+NN, #71, 1-0 (resign)
12. OliThink 5.3.3 vs Phalanx XXV
12) OliThink 5.3.3 vs Phalanx XXV, #96, 0-1 (mate)
13. TSCP 1.81 vs Phalanx XXV
13) TSCP 1.81 vs Phalanx XXV, #114, 0-1 (mate)
14. FrankWalter 2.4.0 vs Phalanx XXV
14) FrankWalter 2.4.0 vs Phalanx XXV, #76, 0-1 (mate)
15. Phalanx XXV vs Scorpio_2.9.0 MCTS+NN
15) Phalanx XXV vs Scorpio_2.9.0 MCTS+NN, #47, 1-0 (timeout)
16. TSCP 1.81 vs OliThink 5.3.3
16) TSCP 1.81 vs OliThink 5.3.3, #72, 0-1 (mate)
17. FrankWalter 2.4.0 vs OliThink 5.3.3
17) FrankWalter 2.4.0 vs OliThink 5.3.3, #97, 1/2-1/2 (repetition)
18. Scorpio_2.9.0 MCTS+NN vs OliThink 5.3.3
18) Scorpio_2.9.0 MCTS+NN vs OliThink 5.3.3, #208, 1/2-1/2 (fifty moves)
19. FrankWalter 2.4.0 vs TSCP 1.81
19) FrankWalter 2.4.0 vs TSCP 1.81, #67, 1-0 (mate)
20. TSCP 1.81 vs Scorpio_2.9.0 MCTS+NN
20) TSCP 1.81 vs Scorpio_2.9.0 MCTS+NN, #144, 0-1 (mate)
21. FrankWalter 2.4.0 vs Scorpio_2.9.0 MCTS+NN
21) FrankWalter 2.4.0 vs Scorpio_2.9.0 MCTS+NN, #71, 1-0 (mate)
-----------------------------------------------------------------------------------------
  #  name                   games    wins%   draws%  losses%    score     los%   elo+/-
  1. Crafty-25.3                6     83.3     16.7      0.0      5.5     98.7    416.6
  2. Phalanx XXV                6     66.7     16.7     16.7      4.5     91.0    190.8
  3. EXchess v7.97b             6     66.7      0.0     33.3      4.0     79.3    120.4
  4. OliThink 5.3.3             6     33.3     33.3     33.3      3.0     50.0     -0.0
  5. FrankWalter 2.4.0          6     33.3     16.7     50.0      2.5     32.7    -58.5
  6. Scorpio_2.9.0 MCTS+NN      6     16.7     16.7     66.7      1.5      9.0   -190.8
  7. TSCP 1.81                  6      0.0      0.0    100.0      0.0      0.0      0.0
-----------------------------------------------------------------------------------------


Tournamemt finished! Elapsed: 2:29
```
